Alignment/HIPAlignmentAlgorithm/plugins/LhcTrackAnalyzer.cc

```diff
@@ -0,0 +1,364 @@
+public:
+  explicit LhcTrackAnalyzer(const edm::ParameterSet&);
+  ~LhcTrackAnalyzer() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  // ----------member data ---------------------------
+  edm::InputTag TrackCollectionTag_;
+  edm::InputTag PVtxCollectionTag_;
+  bool debug_;
+
+  edm::EDGetTokenT<reco::TrackCollection> theTrackCollectionToken;
+  edm::EDGetTokenT<reco::VertexCollection> theVertexCollectionToken;
+
+  // Output
+  std::string filename_;
+  TFile* rootFile_;
+  TTree* rootTree_;
+
+  // Root-Tuple variables :
+  //=======================
+  void SetVarToZero();
+
+  static const int nMaxtracks_ = 3000;
+  int nTracks_;
+  int run_;
+  int event_;
+  double pt_[nMaxtracks_];
+  double eta_[nMaxtracks_];
+  double phi_[nMaxtracks_];
+  double chi2_[nMaxtracks_];
+  double chi2ndof_[nMaxtracks_];
+  int charge_[nMaxtracks_];
+  double qoverp_[nMaxtracks_];
+  double dz_[nMaxtracks_];
+  double dxy_[nMaxtracks_];
+  double xPCA_[nMaxtracks_];
+  double yPCA_[nMaxtracks_];
+  double zPCA_[nMaxtracks_];
+  int trkAlgo_[nMaxtracks_];
+  int trkQuality_[nMaxtracks_];
+  int isHighPurity_[nMaxtracks_];
+  int validhits_[nMaxtracks_][7];
+  bool goodbx_;
+  bool goodvtx_;
+};
+
+// Constructor
+
+LhcTrackAnalyzer::LhcTrackAnalyzer(const edm::ParameterSet& iConfig)
+    : TrackCollectionTag_(iConfig.getParameter<edm::InputTag>("TrackCollectionTag")),
+      PVtxCollectionTag_(iConfig.getParameter<edm::InputTag>("PVtxCollectionTag")),
+      debug_(iConfig.getParameter<bool>("Debug")),
+      filename_(iConfig.getParameter<std::string>("OutputFileName")) {
+  //now do what ever initialization is needed
+  theTrackCollectionToken = consumes<reco::TrackCollection>(TrackCollectionTag_);
+  theVertexCollectionToken = consumes<reco::VertexCollection>(PVtxCollectionTag_);
+}
+
+//
+// member functions
+//
+
+/*****************************************************************************/
+void LhcTrackAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+/*****************************************************************************/
+{
+  using namespace edm;
+  using namespace reco;
+  using namespace std;
+
+  //=======================================================
+  // Initialize Root-tuple variables
+  //=======================================================
+
+  SetVarToZero();
+
+  //=======================================================
+  // Retrieve the Track information
+  //=======================================================
+
+  const auto& vertices = iEvent.get(theVertexCollectionToken);
+  const auto& vtx = vertices.front();
+  if (vtx.isFake()) {
+    goodvtx_ = false;
+  } else {
+    goodvtx_ = true;
+  }
+
+  goodbx_ = true;
+  // int bx = iEvent.bunchCrossing();
+  //if (bx==51 || bx==2724) goodbx_=true;
```

Comment:
Hi @mmusich do we need these commented-out lines? Btw is `goodbx_` ever set to not-true? I didn't see  anything in the code that would do that, so to me it seems this variable is not used. Is that right?